src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/FindPal.cs

```diff
@@ -0,0 +1,268 @@
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+            }
+
+            return findPal.NormalizeOid(maybeOid, oidGroup);
+        }
+
+        private static X509KeyUsageFlags ConfirmedX509KeyUsage(object findValue)
+        {
+            if (findValue is X509KeyUsageFlags)
+                return (X509KeyUsageFlags)findValue;
+
+            if (findValue is int)
+                return (X509KeyUsageFlags)(int)findValue;
+
+            if (findValue is uint)
+                return (X509KeyUsageFlags)(uint)findValue;
+
+            string findValueString = findValue as string;
+
+            if (findValueString != null)
+            {
+                X509KeyUsageFlags usageFlags;
+
+                if (s_keyUsages.TryGetValue(findValueString, out usageFlags))
+                {
+                    return usageFlags;
+                }
+            }
+
+            throw new CryptographicException(SR.Cryptography_X509_InvalidFindValue);
+        }
+
+        //
+        // verify the passed keyValue is valid as per X.208
+        //
+        // The first number must be 0, 1 or 2.
+        // Enforce all characters are digits and dots.
+        // Enforce that no dot starts or ends the Oid, and disallow double dots.
+        // Enforce there is at least one dot separator.
+        //
+        internal static void ValidateOidValue(string keyValue)
+        {
+            if (keyValue == null)
+                throw new ArgumentNullException("keyValue");
+
+            int len = keyValue.Length;
+            if (len < 2)
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+
+            // should not start with a dot. The first digit must be 0, 1 or 2.
+            char c = keyValue[0];
+            if (c != '0' && c != '1' && c != '2')
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+            if (keyValue[1] != '.' || keyValue[len - 1] == '.') // should not end in a dot
+                throw new ArgumentException(SR.Argument_InvalidOidValue);
+
+            // While characters 0 and 1 were both validated, start at character 1 to validate
+            // that there aren't two dots in a row.
+            for (int i = 1; i < len; i++)
+            {
+                // ensure every character is either a digit or a dot
+                if (char.IsDigit(keyValue[i]))
+                    continue;
+                if (keyValue[i] != '.' || keyValue[i + 1] == '.') // disallow double dots
+                    throw new ArgumentException(SR.Argument_InvalidOidValue);
+            }
+        }
+
+        internal static BigInteger PositiveBigIntegerFromByteArray(byte[] bytes)
+        {
+            // To prevent the big integer from misinterpreted as a negative number,
+            // add a "leading 0" to the byte array.
+            // Since BigInteger(bytes[]) requires a little-endian byte array,
+            // the "leading 0" actually goes at the end of the array.
```

Comment:
Because it's already going to be interpreted as positive.  Just ask you asked for back in June :) https://github.com/dotnet/corefx/pull/1990/files#r32145804